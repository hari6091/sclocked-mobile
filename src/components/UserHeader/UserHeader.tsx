import React from "react";
import { Text, HStack, VStack, Icon } from "native-base";
import { FontAwesome } from "@expo/vector-icons";
import { useProfile } from "../../hooks";

const UserHeader = () => {
  const { profile } = useProfile();

  return (
    <HStack
      w="100%"
      h="17%"
      bg="#FFFFFF"
      justifyContent="center"
      alignItems="center"
      space="22px"
      safeArea
    >
      <Icon as={FontAwesome} name="user-circle-o" size="12" color="#28374D" />
      <VStack>
        <Text
          fontSize="18px"
          textAlign="center"
          color="#28374D"
          testID="username"
        >
          {profile?.name ?? "Carregando..."}
        </Text>
        {profile?.matricula && (
          <Text
            fontSize="18px"
            textAlign="center"
            color="#28374D"
            testID="matricula"
          >
            {profile?.matricula ?? "Carregando..."}
          </Text>
        )}
      </VStack>
    </HStack>
  );
};

export default UserHeader;
