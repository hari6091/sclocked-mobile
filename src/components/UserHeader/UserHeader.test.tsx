import React from "react";
import UserHeader from "./UserHeader";
import { render } from "../../test/test-utils";

jest.mock("../../hooks", () => ({
  useProfile: jest.fn(() => ({
    profile: {
      name: "John Doe",
      matricula: "12345",
    },
  })),
}));

describe("UserHeader", () => {
  it("renders without errors", () => {
    const tree = render(<UserHeader />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("displays profile name correctly", () => {
    const { getByTestId, getByText } = render(<UserHeader />);
    const nameElement = getByTestId("username");
    const nameElementText = getByText("John Doe");

    expect(nameElement).toBeTruthy();
    expect(nameElementText).toBeTruthy();
  });

  it("displays profile matricula correctly", () => {
    const { getByTestId, getByText } = render(<UserHeader />);
    const matriculaElement = getByTestId("matricula");
    const matriculaElementText = getByText("12345");

    expect(matriculaElement).toBeTruthy();
    expect(matriculaElementText).toBeTruthy();
  });
});
