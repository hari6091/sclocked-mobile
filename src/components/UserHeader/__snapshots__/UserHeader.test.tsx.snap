// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader renders without errors 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "height": "17%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text />
    <View
      dataSet={{}}
      style={
        {
          "width": 22,
        }
      }
    />
    <View
      dataSet={{}}
      style={{}}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#28374D",
            "fontFamily": undefined,
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 27,
            "textAlign": "center",
            "textDecorationLine": undefined,
          }
        }
        testID="username"
      >
        John Doe
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#28374D",
            "fontFamily": undefined,
            "fontSize": 18,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 27,
            "textAlign": "center",
            "textDecorationLine": undefined,
          }
        }
        testID="matricula"
      >
        12345
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
